design/design-doc-resource-claim-definition.md

```diff
@@ -0,0 +1,174 @@
+    listKind: MySQLInstanceList
+    plural: mysqlinstances
+    singular: mysqlinstance
+  version: v1alpha1
+  validation:
+    openAPIV3Schema:
+      description: A MySQLInstance is a portable resource claim that may be satisfied
+        by binding to a MySQL managed resource such as an AWS RDS instance or a GCP
+        CloudSQL instance.
+      properties:
+        spec:
+          properties:
+            engineVersion:
+              description: EngineVersion specifies the desired MySQL engine version,
+                e.g. 5.7.
+              enum:
+              - "5.6"
+              - "5.7"
+              type: string
+```
+
+```yaml
+---
+apiVersion: core.crossplane.io/v1alpha1
+kind: ResourceBindingDefinition
+metadata:
+  name: mysqlinstance-cloudsqlinstance
+spec:
+  claim:
+    apiVersion: database.crossplane.io/v1alpha1
+    kind: MySQLInstance
+  resources:
+  - resource:
+      apiVersion: database.gcp.crossplane.io/v1beta1
+      kind: CloudSQLInstance
+    class:
+      apiVersion: database.gcp.crossplane.io/v1beta1
+      kind: CloudSQLInstanceClass
+      strategies:
+        provisioning:
+        - overlay
+        - translate
+        overlay:
+        - fromClassField: ".specTemplate"
+          toResourceField: ".spec"
+        translate:
+        - fromClaimField: ".spec.engineVersion"
+          toResourceField: ".spec.forProvider.databaseVersion"
+          transforms:
+            toupper: true
+            replace:
+              string: "."
+              with: "_"
+---
+apiVersion: core.crossplane.io/v1alpha1
+kind: ResourceBindingDefinition
+metadata:
+  name: mysqlinstance-cloudsqlinstance
+spec:
+  claim:
+    apiVersion: database.crossplane.io/v1alpha1
+    kind: MySQLInstance
+  resources:
+  - resource:
+      apiVersion: database.azure.crossplane.io/v1beta1
+      kind: MySQLServer
+    class:
+      apiVersion: database.azure.crossplane.io/v1beta1
+      kind: SQLServerClass
+      strategies:
+        selection:
+        - match
+        match:
+        - fromClaimField: ".spec.engineVersion"
+          toClassField: ".specTemplate.forProvider.version"
+          transforms:
+            toupper: true
+  - resource:
+      apiVersion: database.azure.crossplane.io/v1alpha3
+      kind: MySQLServerVirtualNetworkRule
+    class:
+      apiVersion: database.azure.crossplane.io/v1alpha3
+      kind: MySQLServerVirtualNetworkRuleClass
+```
+
+`ResourceClaimDefinition` is an abstraction that provides guard rails and
+opinions around the process of authoring a `CustomResourceDefinition` that is
+intended for use as a resource claim. Crossplane resource claims are of many
+different kinds, but all have certain things in common. These include:
+
+* Declaring use of the status subresource.
+* Supporting standard Kubernetes object and type metadata.
+* Supporting a binding phase and conditioned status in their `.status`.
+* Supporting a class selector, class reference, resource reference, and
```

Comment:
All these reference interfaces would need to change to accomodate more than one references. It could be helpful to demonstrate how you'd tackle this.

Reply:
Agreed - I'm holding off on putting too much detail here until @bassam has had a chance to weigh in.